src/TeachABit.Front/src/pages/tecajevi_temp/Tecajevi.tsx

```diff
@@ -1,15 +1,80 @@
+import { useEffect, useState } from "react";
+import { TecajDto } from "../../models/TecajDto";
+import requests from "../../api/agent";
+import { Button } from "@mui/material";
+import Tecaj from "./Tecaj";
+import { useGlobalContext } from "../../context/Global.context";
 import SearchBox from "../../components/searchbox/SearchBox";
-
-function abc(query?: string) {
-    console.log("Search:", query);
-}
+import useRequestBuilder from "../../hooks/useRequestBuilder";
 
 export default function Tecajevi() {
+
+    const [tecajList, setTecajList] = useState<TecajDto[]>([]);
+    const globalContext = useGlobalContext();
+
+    const { buildRequest } = useRequestBuilder();
+
+    const GetTecajList = async (search: string | undefined = undefined) => {
+        const response = await requests.getWithLoading(
+            buildRequest("tecajevi", { search })
+        );
+        setTecajList(response.data);
+    };
+
+    useEffect(() => {
+        GetTecajList();
+    }, []);
+
     return (
-        <>
-            <div>
-                <SearchBox onSearch={abc} />
+        <div
+            style={{
+                display: "flex",
+                flexDirection: "column",
+                gap: "20px",
+                alignItems: "flex-start",
+                height: "100%",
+                width: "100%",
+            }}
+        >
+            {globalContext.userIsLoggedIn && (
+                <div
+                    style={{
+                        display: "flex",
+                        flexDirection: "row",
+                        gap: "20px",
+                        width: "100%",
+                        alignItems: "center",
+                    }}
+                >
+                    <SearchBox onSearch={GetTecajList} />
```

Comment:
Aj samo ovo stavi izvan ovog conditiona da je user logged in tak da se moze searchat i dok nisi ulogiran.